vSphere: Crash on minimal configuration

Assuming the following configuration:

```
provider "vsphere" {
  user           = "${var.vsphere_user}"
  password       = "${var.vsphere_password}"
  vcenter_server = "${var.vsphere_vcenter_server}"
}

resource "vsphere_virtual_machine" "web" {
  name   = "terraform_web"
  vcpu   = 2
  memory = 4096
  network_interface {
    label = "10.112.70.0"
  }
  disk {
    size = 40
  }
}
```

Successful plan:

```
➜  terraform git:(master) ✗ terraform plan
Refreshing Terraform state prior to plan...

+ vsphere_virtual_machine.web
    disk.#:                          "" => "1"
    disk.0.size:                     "" => "40"
    domain:                          "" => "vsphere.local"
    memory:                          "" => "4096"
    name:                            "" => "terraform_web"
    network_interface.#:             "" => "1"
    network_interface.0.ip_address:  "" => "<computed>"
    network_interface.0.label:       "" => "10.112.70.0"
    network_interface.0.subnet_mask: "" => "<computed>"
    time_zone:                       "" => "Etc/UTC"
    vcpu:                            "" => "2"


Plan: 1 to add, 0 to change, 0 to destroy.
```

Terraform crash on apply:

```
➜  terraform git:(master) ✗ terraform apply
vsphere_virtual_machine.web: Creating...
  disk.#:                          "" => "1"
  disk.0.size:                     "" => "40"
  domain:                          "" => "vsphere.local"
  memory:                          "" => "4096"
  name:                            "" => "terraform_web"
  network_interface.#:             "" => "1"
  network_interface.0.ip_address:  "" => "<computed>"
  network_interface.0.label:       "" => "10.112.24.0"
  network_interface.0.subnet_mask: "" => "<computed>"
  time_zone:                       "" => "Etc/UTC"
  vcpu:                            "" => "2"
Error applying plan:

1 error(s) occurred:

* vsphere_virtual_machine.web: unexpected EOF

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
panic: runtime error: index out of range
```

VM was successfully created in vSphere after an apply; however, no state was written:

```
➜  terraform git:(master) ✗ cat terraform.tfstate
{
    "version": 1,
    "serial": 0,
    "modules": [
        {
            "path": [
                "root"
            ],
            "outputs": {},
            "resources": {}
        }
    ]
}
```

```
➜  terraform git:(master) ✗ terraform --version
Terraform v0.6.4
```

Full crash log: https://gist.github.com/jwieringa/f0aa0285ed2570c07399

Please let me know if you have any questions. Any feedback on working through the Go stack trace greatly appreciated, thanks!

--- a/builtin/providers/vsphere/resource_vsphere_virtual_machine.go
+++ b/builtin/providers/vsphere/resource_vsphere_virtual_machine.go
@@ -389,6 +389,7 @@ func resourceVSphereVirtualMachineRead(d *schema.ResourceData, meta interface{})
 			networkInterfaces = append(networkInterfaces, networkInterface)
 		}
 	}
+	log.Printf("[DEBUG] networkInterfaces: %#v", networkInterfaces)
 	err = d.Set("network_interface", networkInterfaces)
 	if err != nil {
 		return fmt.Errorf("Invalid network interfaces to set: %#v", networkInterfaces)
@@ -420,10 +421,12 @@ func resourceVSphereVirtualMachineRead(d *schema.ResourceData, meta interface{})
 	d.Set("datastore", rootDatastore)
 
 	// Initialize the connection info
-	d.SetConnInfo(map[string]string{
-		"type": "ssh",
-		"host": networkInterfaces[0]["ip_address"].(string),
-	})
+	if len(networkInterfaces) > 0 {
+		d.SetConnInfo(map[string]string{
+			"type": "ssh",
+			"host": networkInterfaces[0]["ip_address"].(string),
+		})
+	}
 
 	return nil
 }

